### PR TITLE
Added CAPRKE2 tests

### LINKED ISSUE #30: Automate CAPRKE2 tests
- Analyze CAPRKE2  cluster creation with CAPD
- Automate and add tests to cover it

### PR DESCRIPTION
**Pull request by cpinjani**
### What does this PR do?
- Added CAPD+RKE2 tests
- Refactor CAPD tests and adds function for common operations
- Increases timeouts for flaky operations

### Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes:
#30 
and Failing CI

### Checklist:
- [x] GitHub Actions:
https://github.com/rancher/rancher-turtles-e2e/actions/runs/9329300075

### CODE REVIEW

File: tests/cypress/latest/fixtures/capd-rke2-provider.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: turtles-capi.cattle.io/v1alpha1
+kind: CAPIProvider
+metadata:
+  name: docker-rke2
+  namespace: capd-system
+spec:
+  name: docker-rke2
+  type: infrastructure
+  configSecret:
+    name: variables
```
[the remark this answers is not in the document]
Comment (cpinjani) [reply tests/cypress/latest/fixtures/capd-rke2-provider.yaml:10]: Docker provider is created using CAPI UI so has unique name each time, not conflicting with this name

File: tests/cypress/latest/e2e/unit_tests/capd_cluster.spec.ts [hunk 2]
```diff
@@ -55,51 +86,52 @@ describe('Import CAPD', () => {
...
-        cy.checkCAPICluster(clusterShort);
+        // Install App
+        cy.installApp('Monitoring', 'cattle-monitoring');
       })
-    );
+    }
+
+    if (path == 'cluster_autoimport') {
+      qase(12,
+        it('Scale imported CAPD cluster', () => {
```
[the remark this answers is not in the document]
Comment (cpinjani) [reply tests/cypress/latest/e2e/unit_tests/capd_cluster.spec.ts]: Resolved

File: tests/cypress/latest/e2e/unit_tests/capd_cluster.spec.ts [hunk 3]
```diff
@@ -55,51 +86,52 @@ describe('Import CAPD', () => {
...
-      // Click on imported CAPD cluster
-      cy.contains(clusterFull).click();
-
-      // Install App
-      cy.installApp('Monitoring', 'cattle-monitoring');
-    })
-
-    qase(12,
-      it('Scale imported CAPD cluster', () => {
+    if (path.includes('namespace_autoimport')) {
```
[the remark this answers is not in the document]
Comment (cpinjani) [reply tests/cypress/latest/e2e/unit_tests/capd_cluster.spec.ts]: Resolved

**Comment by cpinjani** [comment]
https://github.com/rancher/rancher-turtles-e2e/actions/runs/9362576713

**Review by valaparthvi — APPROVED** [review]

**Pull request merged**